File: frontend/lib/models/menu_item.dart
// MenuItem class - represents food items in menu
// Like pizza, burger, etc.

class MenuItem {
  String id;
  String name;
  String description;
  double price;
  bool isVeg;
  String category;
  bool isAvailable;
  String image; // URL of food image

  // Constructor
  MenuItem({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.isVeg,
    required this.category,
    required this.isAvailable,
    required this.image,
  });

  // Convert JSON to MenuItem object
  factory MenuItem.fromJson(Map<String, dynamic> json) {
    return MenuItem(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      price: (json['price'] ?? 0).toDouble(),
      isVeg: json['isVeg'] ?? true,
      category: json['category'] ?? 'Other',
      isAvailable: json['isAvailable'] ?? true,
      image: json['image'] ?? 'https://via.placeholder.com/300x200?text=Food+Item',
    );
  }
}

File: backend/src/models/MenuItem.js
// MenuItem model - represents food items in restaurant menus
// Each item belongs to one restaurant

import mongoose from "mongoose";

// Define how a MenuItem looks in the database
const menuItemSchema = new mongoose.Schema(
  {
    // Which restaurant owns this item
    restaurantId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Restaurant",
      required: true
    },

    // Name of the food item
    name: {
      type: String,
      required: true,
      trim: true
    },

    // Description of the dish
    description: {
      type: String,
      default: ""
    },

    // Price in your currency (e.g., ₹)
    price: {
      type: Number,
      required: true,
      min: 0
    },

    // Is it vegetarian?
    isVeg: {
      type: Boolean,
      default: true
    },

    // Category (e.g., "Starters", "Main Course", "Desserts")
    category: {
      type: String,
      enum: [
        "Appetizers",
        "Main Course", 
        "Desserts",
        "Beverages",
        "Breads",
        "Rice & Biryani",
        "Chinese",
        "South Indian",
        "Fast Food",
        "Salads",
        "Other"
      ],
      default: "Other"
    },

    // Image URL for the food item
    image: {
      type: String,
      default: "https://via.placeholder.com/300x200?text=Food+Item"
    },

    // Is this item available for ordering right now?
    isAvailable: {
      type: Boolean,
      default: true
    }
  },
  {
    // Automatically add createdAt and updatedAt
    timestamps: true
  }
);

// Create index for finding items by restaurant
menuItemSchema.index({ restaurantId: 1 });

// Create the MenuItem model
const MenuItem = mongoose.model("MenuItem", menuItemSchema);

export default MenuItem;
